files/en-us/web/javascript/reference/global_objects/asyncgeneratorfunction/index.html

```diff
@@ -0,0 +1,114 @@
+---
+title: AsyncGeneratorFunction
+slug: Web/JavaScript/Reference/Global_Objects/AsyncGeneratorFunction
+tags:
+- Constructor
+- ECMAScript 2018
+- AsyncGeneratorFunction
+- Iterator
+- JavaScript
+- Reference
+---
+<div>{{JSRef}}</div>
+
+<p>The <strong><code>AsyncGeneratorFunction</code> constructor</strong> creates a new
+  {{jsxref("Statements/async_function*", "async generator function", "", 1)}} object. In JavaScript,
+  every async generator function is actually an <code>AsyncGeneratorFunction</code> object.</p>
+
+<p>Note that <code>AsyncGeneratorFunction</code> is not a global object. It could be obtained
+  by evaluating the following code.</p>
+
+<pre class="brush: js">Object.getPrototypeOf(async function*(){}).constructor
+</pre>
+
+<h2 id="Syntax">Syntax</h2>
+
+<pre
+  class="brush: js">new AsyncGeneratorFunction ([<var>arg1</var>[, <var>arg2</var>[, ...<var>argN</var>]],] <var>functionBody</var>)</pre>
+
+<h3 id="Parameters">Parameters</h3>
+
+<dl>
+  <dt><code>arg1, arg2, ... arg<em>N</em></code></dt>
+  <dd>Names to be used by the function as formal argument names. Each must be a string
+    that corresponds to a valid JavaScript identifier or a list of such strings separated
+    with a comma; for example "<code>x</code>", "<code>theValue</code>", or
+    "<code>a,b</code>".</dd>
+  <dt><code>functionBody</code></dt>
+  <dd>A string containing the JavaScript statements comprising the function definition.
+  </dd>
+</dl>
+
+<h2 id="Description">Description</h2>
+
+<p>{{jsxref("Statements/async_function*", "async generator function")}} objects created with the
+  <code>AsyncGeneratorFunction</code> constructor are parsed when the function is created. This
+  is less efficient than declaring a generator function with an
+  {{jsxref("Statements/async_function*", "async function* expression")}} and calling it within your
+  code, because such functions are parsed with the rest of the code.</p>
+
+<p>All arguments passed to the function are treated as the names of the identifiers of the
+  parameters in the function to be created, in the order in which they are passed.</p>
+
+<div class="note">
```

Comment:
This will need another class `notecard`.
```suggestion
<div class="notecard note">
```